client/src/components/Settings.vue

```diff
@@ -4,24 +4,45 @@
     type='number'
     min='1' max='6'
     name='num_players'
-    v-model="numPlayers"/>
+    v-model='numPlayers'/>
 
+    <DisplayNames ref='names' v-model='numPlayers'/>
+    <br/>
     <button @click="submit">Submit</button>
   </div>
 </template>
 
 <script>
 import { settingsStore } from '../store/settings.js'
+import { playerChoiceStore } from '../store/player-choices.js'
+import DisplayNames from './Settings-DisplayNames.vue'
+import uuid from 'uuid/v4'
+
 export default {
   data () {
     return {
       numPlayers: 2
     }
   },
+  components: {
+    DisplayNames
+  },
   methods: {
     submit () {
       settingsStore.commit('setNumPlayers', this.numPlayers)
-      this.$router.push('pick')
+
+      // register all players when number has been set
+      for (let i of this.$refs.names.names) {
```

Comment:
While there's nothing wrong with for loops, the js community tends to lean more to using forEach.
Also, there should be a better name than i